src/client/caching/structures/channels/GuildTextChannel.ts

```diff
@@ -49,13 +51,23 @@ export abstract class GuildTextChannel extends GuildChannel {
 
 	/**
 	 * Sends a message to the channel.
-	 * @param content The {@link MessageBuilder builder} to send.
+	 * @param data The {@link MessageBuilder builder} to send.
 	 * @since 0.0.1
 	 */
-	public async send(content: MessageBuilder): Promise<this> {
-		const data = await this.client.api.post(Routes.channelMessages(this.id), content);
+	public async send(data: MessageOptions, splitOptions: SplitOptions): Promise<Message[]>
+	public async send(data: (message: MessageBuilder) => MessageBuilder, splitOptions: SplitOptions): Promise<Message[]>
+	public async send(data: MessageOptions | ((message: MessageBuilder) => MessageBuilder), splitOptions: SplitOptions): Promise<Message[]> {
+		const split = new MessageBuilder(typeof data === 'function' ? data(new MessageBuilder()) : data).split(splitOptions);
+
+		const endpoint = Routes.channelMessages(this.id);
+		const responses = [];
+
+		for (const message of split) responses.push(this.client.api.post(endpoint, message));
+
+		const rawMessages = await Promise.all(responses);
+
 		// eslint-disable-next-line dot-notation
-		return new this.guild.channels['Holds'](this.client, data, this.guild) as this;
+		return rawMessages.map(msg => this.messages['_add'](msg as APIMessageData));
```

Comment:
Those shouldn't be added to the cache, the actions do that.

Reply:
They absolutely must return an instance that can be relied upon. All rest methods that return instances need to have consistent references.

Command Editing and more relies upon this basic requirement.

Reply:
You can at least use Holds to build new instances (to skip cache checks in the actions), or pass `false` as second argument so it doesn't get stored in the cache.

Reply:
Caching is required for Command Editing to work. That's what I mean by maintaining a consistent reference. However, I included an option to skip caching. But this is going to have to be an option on practically all of our rest methods now, which sucks imo.